### ISSUE TITLE
When i use Polygraphy to compare the accuracy between trt and onnx, there is a weird accuracy lost. I don’t choose either fp16 or int8. I want to know how to figure it out and why it occurs?

### ISSUE THREAD

**Issue by tianyashu2020** [NVIDIA/TensorRT#2895]
## Environment

**TensorRT Version**: TensorRT-8.6.0.12
**NVIDIA GPU**: GPU 0: NVIDIA GeForce RTX 4090 (UUID: GPU-332270d0-b8bb-d083-ebc4-3d26893d5aa2)
**NVIDIA Driver Version**: 
**CUDA Version**: 11.6
**CUDNN Version**: 
**Operating System**: ubuntu 20.04
**Python Version (if applicable)**: 3.9
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

[I] RUNNING | Command: ./polygraphy run flownoise_sim.onnx --trt --onnxrt
[I] trt-runner-N0-04/21/23-18:46:01     | Activating and starting inference
[W] CUDA lazy loading is not enabled. Enabling it can significantly reduce device memory usage and speed up TensorRT initialization. See "Lazy Loading" section of CUDA documentation https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#lazy-loading
[W] onnx2trt_utils.cpp:374: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[W] onnx2trt_utils.cpp:400: One or more weights outside the range of INT32 was clamped
[I]     Configuring with profiles: [Profile().add('input_1', min=[1, 1, 400], opt=[1, 1, 400], max=[1, 1, 400]).add('h_in1', min=[1, 1, 256], opt=[1, 1, 256], max=[1, 1, 256]).add('c_in1', min=[1, 1, 256], opt=[1, 1, 256], max=[1, 1, 256]).add('h_in2', min=[1, 1, 256], opt=[1, 1, 256], max=[1, 1, 256]).add('c_in2', min=[1, 1, 256], opt=[1, 1, 256], max=[1, 1, 256])]
[I] Building engine with configuration:
    Flags                  | []
    Engine Capability      | EngineCapability.DEFAULT
    Memory Pools           | [WORKSPACE: 24217.31 MiB, TACTIC_DRAM: 24217.31 MiB]
    Tactic Sources         | [CUBLAS, CUBLAS_LT, CUDNN, EDGE_MASK_CONVOLUTIONS, JIT_CONVOLUTIONS]
    Profiling Verbosity    | ProfilingVerbosity.DETAILED
    Preview Features       | [FASTER_DYNAMIC_SHAPES_0805, DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
[I] Finished engine building in 82.442 seconds
[I] trt-runner-N0-04/21/23-18:46:01    
[I] trt-runner-N0-04/21/23-18:46:01    
    ---- Inference Input(s) ----
    {input_1 [dtype=float32, shape=(1, 1, 400)],
     h_in1 [dtype=float32, shape=(1, 1, 256)],
     c_in1 [dtype=float32, shape=(1, 1, 256)],
     h_in2 [dtype=float32, shape=(1, 1, 256)],
     c_in2 [dtype=float32, shape=(1, 1, 256)]}
[I] trt-runner-N0-04/21/23-18:46:01    
    ---- Inference Output(s) ----
    {h_out1 [dtype=float32, shape=(1, 1, 256)],
     c_out1 [dtype=float32, shape=(1, 1, 256)],
     h_out2 [dtype=float32, shape=(1, 1, 256)],
     c_out2 [dtype=float32, shape=(1, 1, 256)],
     output_1 [dtype=float32, shape=(1, 400)]}
[I] trt-runner-N0-04/21/23-18:46:01     | Completed 1 iteration(s) in 1.298 ms | Average inference time: 1.298 ms.
[I] onnxrt-runner-N0-04/21/23-18:46:01  | Activating and starting inference
[I] Creating ONNX-Runtime Inference Session with providers: ['CPUExecutionProvider']
[I] onnxrt-runner-N0-04/21/23-18:46:01 
    ---- Inference Input(s) ----
    {input_1 [dtype=float32, shape=(1, 1, 400)],
     h_in1 [dtype=float32, shape=(1, 1, 256)],
     c_in1 [dtype=float32, shape=(1, 1, 256)],
     h_in2 [dtype=float32, shape=(1, 1, 256)],
     c_in2 [dtype=float32, shape=(1, 1, 256)]}
[I] onnxrt-runner-N0-04/21/23-18:46:01 
    ---- Inference Output(s) ----
    {output_1 [dtype=float32, shape=(1, 400)],
     h_out1 [dtype=float32, shape=(1, 1, 256)],
     c_out1 [dtype=float32, shape=(1, 1, 256)],
     h_out2 [dtype=float32, shape=(1, 1, 256)],
     c_out2 [dtype=float32, shape=(1, 1, 256)]}
[I] onnxrt-runner-N0-04/21/23-18:46:01  | Completed 1 iteration(s) in 1.386 ms | Average inference time: 1.386 ms.
[I] Accuracy Comparison | trt-runner-N0-04/21/23-18:46:01 vs. onnxrt-runner-N0-04/21/23-18:46:01
[I]     Comparing Output: 'h_out1' (dtype=float32, shape=(1, 1, 256)) with 'h_out1' (dtype=float32, shape=(1, 1, 256))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-04/21/23-18:46:01: h_out1 | Stats: mean=0.094774, std-dev=0.59072, var=0.34895, median=0, min=-1 at (0, 0, 215), max=1 at (0, 0, 62), avg-magnitude=0.41476
[I]             ---- Histogram ----
                Bin Range            |  Num Elems | Visualization
                (-1      , -0.8    ) |         28 | #########
                (-0.8    , -0.6    ) |         16 | #####
                (-0.6    , -0.4    ) |          0 | 
                (-0.4    , -0.2    ) |          5 | #
                (-0.2    , 1.34e-06) |        116 | ########################################
                (1.34e-06, 0.2     ) |         11 | ###
                (0.2     , 0.4     ) |          5 | #
                (0.4     , 0.6     ) |          6 | ##
                (0.6     , 0.8     ) |         28 | #########
                (0.8     , 1       ) |         41 | ##############
[I]         onnxrt-runner-N0-04/21/23-18:46:01: h_out1 | Stats: mean=0.094774, std-dev=0.59072, var=0.34895, median=0, min=-1 at (0, 0, 215), max=1 at (0, 0, 62), avg-magnitude=0.41476
[I]             ---- Histogram ----
                Bin Range            |  Num Elems | Visualization
                (-1      , -0.8    ) |         28 | #########
                (-0.8    , -0.6    ) |         16 | #####
                (-0.6    , -0.4    ) |          0 | 
                (-0.4    , -0.2    ) |          5 | #
                (-0.2    , 1.34e-06) |        116 | ########################################
                (1.34e-06, 0.2     ) |         11 | ###
                (0.2     , 0.4     ) |          5 | #
                (0.4     , 0.6     ) |          6 | ##
                (0.6     , 0.8     ) |         28 | #########
                (0.8     , 1       ) |         41 | ##############
[I]         Error Metrics: h_out1
[I]             Minimum Required Tolerance: elemwise error | [abs=2.6226e-05] OR [rel=nan] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=4.6509e-07, std-dev=2.1569e-06, var=4.6522e-12, median=2.4515e-32, min=0 at (0, 0, 2), max=2.6226e-05 at (0, 0, 146), avg-magnitude=4.6509e-07
[I]                 ---- Histogram ----
                    Bin Range            |  Num Elems | Visualization
                    (0       , 2.62e-06) |        246 | ########################################
                    (2.62e-06, 5.25e-06) |          3 | 
                    (5.25e-06, 7.87e-06) |          3 | 
                    (7.87e-06, 1.05e-05) |          2 | 
                    (1.05e-05, 1.31e-05) |          0 | 
                    (1.31e-05, 1.57e-05) |          1 | 
                    (1.57e-05, 1.84e-05) |          0 | 
                    (1.84e-05, 2.1e-05 ) |          0 | 
                    (2.1e-05 , 2.36e-05) |          0 | 
                    (2.36e-05, 2.62e-05) |          1 | 
[I]             Relative Difference | Stats: mean=nan, std-dev=nan, var=nan, median=nan, min=nan at (0, 0, 7), max=nan at (0, 0, 7), avg-magnitude=nan
[I]                 
[E]         FAILED | Output: 'h_out1' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'c_out1' (dtype=float32, shape=(1, 1, 256)) with 'c_out1' (dtype=float32, shape=(1, 1, 256))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-04/21/23-18:46:01: c_out1 | Stats: mean=0.15519, std-dev=2.63, var=6.9171, median=4.5215e-24, min=-6.5663 at (0, 0, 215), max=7.0916 at (0, 0, 62), avg-magnitude=1.7508
[I]             ---- Histogram ----
                Bin Range      |  Num Elems | Visualization
                (-6.57, -5.2 ) |          9 | ###
                (-5.2 , -3.83) |         17 | ######
                (-3.83, -2.47) |         10 | ###
                (-2.47, -1.1 ) |          7 | ##
                (-1.1 , 0.263) |        110 | ########################################
                (0.263, 1.63 ) |         54 | ###################
                (1.63 , 2.99 ) |         19 | ######
                (2.99 , 4.36 ) |          9 | ###
                (4.36 , 5.73 ) |         11 | ####
                (5.73 , 7.09 ) |         10 | ###
[I]         onnxrt-runner-N0-04/21/23-18:46:01: c_out1 | Stats: mean=0.15519, std-dev=2.63, var=6.9171, median=0, min=-6.5663 at (0, 0, 215), max=7.0916 at (0, 0, 62), avg-magnitude=1.7508
[I]             ---- Histogram ----
                Bin Range      |  Num Elems | Visualization
                (-6.57, -5.2 ) |          9 | ###
                (-5.2 , -3.83) |         17 | ######
                (-3.83, -2.47) |         10 | ###
                (-2.47, -1.1 ) |          7 | ##
                (-1.1 , 0.263) |        110 | ########################################
                (0.263, 1.63 ) |         54 | ###################
                (1.63 , 2.99 ) |         19 | ######
                (2.99 , 4.36 ) |          9 | ###
                (4.36 , 5.73 ) |         11 | ####
                (5.73 , 7.09 ) |         10 | ###
[I]         Error Metrics: c_out1
[I]             Minimum Required Tolerance: elemwise error | [abs=3.2425e-05] OR [rel=nan] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=1.3194e-06, std-dev=4.0791e-06, var=1.6639e-11, median=1.8517e-09, min=0 at (0, 0, 0), max=3.2425e-05 at (0, 0, 107), avg-magnitude=1.3194e-06
[I]                 ---- Histogram ----
                    Bin Range            |  Num Elems | Visualization
                    (0       , 3.24e-06) |        233 | ########################################
                    (3.24e-06, 6.48e-06) |          7 | #
                    (6.48e-06, 9.73e-06) |          6 | #
                    (9.73e-06, 1.3e-05 ) |          4 | 
                    (1.3e-05 , 1.62e-05) |          1 | 
                    (1.62e-05, 1.95e-05) |          2 | 
                    (1.95e-05, 2.27e-05) |          0 | 
                    (2.27e-05, 2.59e-05) |          0 | 
                    (2.59e-05, 2.92e-05) |          1 | 
                    (2.92e-05, 3.24e-05) |          2 | 
[I]             Relative Difference | Stats: mean=nan, std-dev=nan, var=nan, median=nan, min=nan at (0, 0, 7), max=nan at (0, 0, 7), avg-magnitude=nan
[I]                 
[E]         FAILED | Output: 'c_out1' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'h_out2' (dtype=float32, shape=(1, 1, 256)) with 'h_out2' (dtype=float32, shape=(1, 1, 256))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-04/21/23-18:46:01: h_out2 | Stats: mean=-0.019106, std-dev=0.50288, var=0.25289, median=-0.038225, min=-0.98826 at (0, 0, 206), max=0.97418 at (0, 0, 234), avg-magnitude=0.41231
[I]         onnxrt-runner-N0-04/21/23-18:46:01: h_out2 | Stats: mean=-0.019105, std-dev=0.50288, var=0.25289, median=-0.038224, min=-0.98826 at (0, 0, 206), max=0.97418 at (0, 0, 234), avg-magnitude=0.41231
[I]         Error Metrics: h_out2
[I]             Minimum Required Tolerance: elemwise error | [abs=5.2154e-06] OR [rel=0.0010483] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=7.4031e-07, std-dev=8.1851e-07, var=6.6997e-13, median=4.7684e-07, min=0 at (0, 0, 46), max=5.2154e-06 at (0, 0, 39), avg-magnitude=7.4031e-07
[I]             Relative Difference | Stats: mean=9.7035e-06, std-dev=6.6439e-05, var=4.4141e-09, median=1.4362e-06, min=0 at (0, 0, 46), max=0.0010483 at (0, 0, 238), avg-magnitude=9.7035e-06
[I]         PASSED | Output: 'h_out2' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'c_out2' (dtype=float32, shape=(1, 1, 256)) with 'c_out2' (dtype=float32, shape=(1, 1, 256))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-04/21/23-18:46:01: c_out2 | Stats: mean=-0.04129, std-dev=1.0256, var=1.0518, median=-0.057686, min=-3.4769 at (0, 0, 168), max=3.5751 at (0, 0, 234), avg-magnitude=0.77868
[I]         onnxrt-runner-N0-04/21/23-18:46:01: c_out2 | Stats: mean=-0.041289, std-dev=1.0256, var=1.0518, median=-0.057685, min=-3.4769 at (0, 0, 168), max=3.5751 at (0, 0, 234), avg-magnitude=0.77868
[I]         Error Metrics: c_out2
[I]             Minimum Required Tolerance: elemwise error | [abs=8.9407e-06] OR [rel=0.0010494] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=1.5511e-06, std-dev=1.384e-06, var=1.9155e-12, median=1.0729e-06, min=0 at (0, 0, 138), max=8.9407e-06 at (0, 0, 5), avg-magnitude=1.5511e-06
[I]             Relative Difference | Stats: mean=1.0028e-05, std-dev=6.6479e-05, var=4.4195e-09, median=1.8879e-06, min=0 at (0, 0, 138), max=0.0010494 at (0, 0, 238), avg-magnitude=1.0028e-05
[I]         PASSED | Output: 'c_out2' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'output_1' (dtype=float32, shape=(1, 400)) with 'output_1' (dtype=float32, shape=(1, 400))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-04/21/23-18:46:01: output_1 | Stats: mean=-0.1004, std-dev=0.13298, var=0.017684, median=-0.11779, min=-0.2794 at (0, 160), max=0.27238 at (0, 0), avg-magnitude=0.15045
[I]         onnxrt-runner-N0-04/21/23-18:46:01: output_1 | Stats: mean=-0.1004, std-dev=0.13298, var=0.017684, median=-0.11779, min=-0.2794 at (0, 160), max=0.27238 at (0, 0), avg-magnitude=0.15045
[I]         Error Metrics: output_1
[I]             Minimum Required Tolerance: elemwise error | [abs=5.6624e-07] OR [rel=2.3877e-05] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=1.4808e-07, std-dev=1.1629e-07, var=1.3524e-14, median=1.1921e-07, min=0 at (0, 1), max=5.6624e-07 at (0, 322), avg-magnitude=1.4808e-07
[I]             Relative Difference | Stats: mean=1.4465e-06, std-dev=2.1462e-06, var=4.606e-12, median=9.0796e-07, min=0 at (0, 1), max=2.3877e-05 at (0, 74), avg-magnitude=1.4465e-06
[I]         PASSED | Output: 'output_1' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[E]     FAILED | Mismatched outputs: ['h_out1', 'c_out1']
[E] Accuracy Summary | trt-runner-N0-04/21/23-18:46:01 vs. onnxrt-runner-N0-04/21/23-18:46:01 | Passed: 0/1 iterations | Pass Rate: 0.0%
[E] FAILED | Runtime: 85.732s | Command: ./polygraphy run flownoise_sim.onnx --trt --onnxrt

**Comment by zerollzeng** [comment]
I can see the Absolute Difference is very small, try `./polygraphy run flownoise_sim.onnx --trt --onnxrt --atol 0.001 --rtol 0.001`

**Comment by tianyashu2020** [comment]
> polygraphy has strict diff check by default
yes,you are right,I find the  Absolute Difference almost has no bad effect  on my case，
very thanks.

**Comment by zerollzeng** [comment]
Can we close this issue? Thanks!

**Comment by tianyashu2020** [comment]
> Can we close this issue? Thanks!
yes，of course. very thanks.
